lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -229,18 +226,42 @@ bool ProxyRenderDelegate::requiresUpdate(const MSubSceneContainer& container, co
     return true;
 }
 
+void ProxyRenderDelegate::_ClearRenderDelegate()
+{
+    delete _sceneDelegate;
+    _sceneDelegate = nullptr;
+    delete _taskController;
+    _taskController = nullptr;
+    delete _renderIndex;
+    _renderIndex = nullptr;
+    delete _renderDelegate;
+    _renderDelegate = nullptr;
+}
+
 //! \brief  One time initialization of this drawing routine
-void ProxyRenderDelegate::_InitRenderDelegate() {
-    // No need to run all the checks if we got till the end
-    if (_isInitialized())
-        return;
+void ProxyRenderDelegate::_InitRenderDelegate(MSubSceneContainer& container) {
 
     if (_proxyShape == nullptr)
         return;
 
-    if (!_usdStage) {
-        _usdStage = _proxyShape->getUsdStage();
+    UsdStageRefPtr newUsdStage = _proxyShape->getUsdStage();
+    if (_usdStage != newUsdStage)
+    {
+        _ClearRenderDelegate();
+
+        // delete everything so we stop drawing the old stage and draw the new one
+        _ClearRenderDelegate();
```

Comment:
you don't have to do it twice...see line 250.

Reply:
Oops, yes once should be enough. I made the helper method because the deletion order matters, if we use the wrong order we can crash deleting the renderIndex. I ended up copying the ordering from the destructor and that seems to work fine.